Add Armor King and Miary Zo

suprovsky opened the pull request:
if u want i can add a pipeline building the container automatically on push

suprovsky committed:
add Armor King

suprovsky committed:
add Dockerfile which is distroless and rootless

TLNBS2405 commented:
Can you do a pre commit check for the test to be green?

suprovsky committed:
update pre-commit-config

suprovsky committed:
add build pipeline for docker releases, add container scanning and checking docker formatting

suprovsky committed:
remove redundant tag for a docker image

suprovsky committed:
fix pre-commit check

suprovsky committed:
fix hadolint issues

suprovsky committed:
add armor king data for unit tests

suprovsky committed:
add a missing line in armor_king.json

suprovsky committed:
fix NUM_CHARACTERS

suprovsky committed:
add workdir so hadolint shuts up

suprovsky committed:
update python-app pipeline dependencies and switch to python 3.11

suprovsky committed:
remove deprecation warning

suprovsky commented:
i've also added:
- github ci pipelines for building, scanning and checking Dockerfile quality (you must add a secret called `GHCR_TOKEN` to the repo, otherwise pushing the image to github will fail, also after you do that you have to make the image public, so it's possible to download it for others
- distroless container uses python 3.11, so i updated python-app.yml, so it has up to date versions of modules, made it use python 3.11 (it works, i'm running the bot on my own server now)
- updated pre-commit-config, so there is no deprecation warning of a stage name being run

if u want to talk quicker on chat or have quick questions, i joined wavu discord as suprovsky, so feel free to dm me

suprovsky committed:
update readme example for config.json, so it's less misleading

AbhijeetKrishnan commented:
Could you split this into two PRs? One to add Armor King, and the other to dockerize the app. I'm also not sure what benefit we'd get from the latter.

suprovsky commented:
sure will do, having app containerized lets you host an app in a container system like Docker or Kubernetes and that gives a lot of advantages compared to standard hosting, you should have a look at these in the Internet to know more, but for me the biggest selling point is separation between apps and not dealing with multiple versions of the same binaries/libraries on a host

suprovsky committed:
remove docker stuff

suprovsky committed:
add Miary Zo

suprovsky commented:
updated, i've added Miary Zo, because it makes sense to have two the same type of features in one pull request
when merging, please just squash it, so it doesn't contain trash from my commits

suprovsky committed:
return pre-commit check

suprovsky committed:
add moves for Miary Zo unit tests

suprovsky committed:
add missing newline
